apps/src/blockly/constants.js

```diff
@@ -50,3 +51,22 @@ export const DEFAULT_SOUND = 'sound://category_digital/ping.mp3';
 export const NO_OPTIONS_MESSAGE = 'uninitialized';
 export const EMPTY_OPTION = '???';
 export const WORKSPACE_PADDING = 16;
+
+export function stringIsXml(str) {
+  try {
+    JSON.parse(str);
```

Comment:
is there a reason you went with this option over checking the first character?

Reply:
Oh, yes there is is. I was worried that in some cases, our saved XML might actually have some empty spaces or even "/n" at the start. Checking the first character didn't feel reliable enough.